pallets/passkey/src/lib.rs

```diff
@@ -127,53 +131,115 @@ pub mod module {
 	impl<T: Config> ValidateUnsigned for Pallet<T> {
 		type Call = Call<T>;
 		fn validate_unsigned(_source: TransactionSource, call: &Self::Call) -> TransactionValidity {
-			Self::validate_signatures(call)?;
-			Ok(ValidTransaction::default())
+			let payload = Self::filter_valid_calls(&call)?;
+			Self::validate_signatures(&payload)?;
+			Self::validate_nonce(&payload)
+		}
+
+		fn pre_dispatch(call: &Self::Call) -> Result<(), TransactionValidityError> {
+			Self::validate_unsigned(TransactionSource::InBlock, call)?;
+
+			let payload = Self::filter_valid_calls(&call)?;
+			Self::pre_dispatch_nonce(&payload)
 		}
 	}
+}
 
-	impl<T: Config> Pallet<T> {
-		fn validate_signatures(call: &Call<T>) -> TransactionValidity {
-			match call {
-				Call::proxy { payload } => {
-					let signed_data = payload.passkey_public_key;
-					let signature = payload.passkey_call.account_ownership_proof.clone();
-					let signer = &payload.passkey_call.account_id;
-					match Self::check_account_signature(signer, &signed_data.into(), &signature) {
-						Ok(_) => Ok(ValidTransaction::default()),
-						Err(_e) => InvalidTransaction::BadSigner.into(),
-					}
-				},
-				_ => InvalidTransaction::Call.into(),
-			}
+impl<T: Config> Pallet<T> {
+	fn filter_valid_calls(call: &Call<T>) -> Result<PasskeyPayload<T>, TransactionValidityError> {
+		match call {
+			Call::proxy { payload }
+				if T::PasskeyCallFilter::contains(&payload.clone().passkey_call.call) =>
+				return Ok(payload.clone()),
```

Comment:
I've been away from Rust for too long... didn't realize you could do this (embed an "if" into a match arm)